Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1352 @@
+            var requests = new List<HistoryRequest>();
+            requests.Add(request);
+
+            foreach (var historyRequest in requests)
+            {
+                if (historyRequest.Symbol.ID.SecurityType != SecurityType.Equity && historyRequest.Symbol.ID.SecurityType != SecurityType.Future && historyRequest.Symbol.ID.SecurityType != SecurityType.Option)
+                {
+                    throw new ArgumentException("Zerodha does not support this security type: " + historyRequest.Symbol.ID.SecurityType);
+                }
+
+                if (historyRequest.StartTimeUtc >= historyRequest.EndTimeUtc)
+                {
+                    throw new ArgumentException("Invalid date range specified");
+                }
+
+                var start = historyRequest.StartTimeUtc.ConvertTo(DateTimeZone.Utc, TimeZones.Kolkata);
+                var end = historyRequest.EndTimeUtc.ConvertTo(DateTimeZone.Utc, TimeZones.Kolkata);
+
+                var history = Enumerable.Empty<BaseData>();
+
+                if (Config.GetBool("zerodha-history-subscription"))
+                {
+                    switch (historyRequest.Resolution)
+                    {
+                        case Resolution.Minute:
+                            history = GetHistoryForPeriod(historyRequest.Symbol, start, end, historyRequest.Resolution, "minute");
+                            break;
+
+                        case Resolution.Hour:
+                            history = GetHistoryForPeriod(historyRequest.Symbol, start, end, historyRequest.Resolution, "60minute");
+                            break;
+
+                        case Resolution.Daily:
+                            history = GetHistoryForPeriod(historyRequest.Symbol, start, end, historyRequest.Resolution, "day");
+                            break;
+                    }
+                }
+
+                foreach (var baseData in history)
+                {
+                    yield return baseData;
+                }
+            }
+        }
+
+        private IEnumerable<BaseData> GetHistoryForPeriod(Symbol symbol, DateTime start, DateTime end, Resolution resolution, string zerodhaResolution)
+        {
+            Log.Debug("ZerodhaBrokerage.GetHistoryForPeriod();");
+            var scripSymbol = _symbolMapper.GetZerodhaInstrumentToken(symbol.Value, symbol.ID.Market);
+            var candles = _kite.GetHistoricalData(scripSymbol.ToStringInvariant(), start, end, zerodhaResolution);
+
+            if (!candles.Any())
+            {
+                OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, "NoHistoricalData",
+                    $"Exchange returned no data for {symbol} on history request " +
+                    $"from {start:s} to {end:s}"));
+            }
+
+            var period = resolution.ToTimeSpan();
+
+            foreach (var candle in candles)
+            {
+                yield return new TradeBar()
```

Comment:
Believe the caller is missing to double check the requested type is `TradeBars`, for example for equities QuoteBars subscriptions are also present